### PR TITLE
rvsdg2dot use argument and result index as label

### LINKED ISSUE #626: Confusing naming of arguments and results in RVSDG dot graph
The arguments and results of a gamma node in an RVSDG dot graph are named confusing:

image

The corresponding arguments in graph10 and graph11 above represent the same variable (a19 and a21 represent the same variable, as well as a20 and a22), but are named differently. I would have expected the same name in graph10 and graph11. 

A similar problem exists for the results. I would have expected that r6 and r7 is named the same.

### PR DESCRIPTION
**Pull request by haved**
In order to make it easier to read the graphs, arguments and results only get their index. The number in parenthesis is the index of the corresponding input/output of the structural node if any.

image

Let me know if you prefer a different label.
Fixes #626

### CODE REVIEW

**Comment by haved** [comment]
Here is an alternative label format. Let me know which you prefer:

image

File: jlm/llvm/backend/dot/DotWriter.cpp [hunk 1]
```diff
@@ -144,9 +144,17 @@ CreateGraphNodes(util::Graph & graph, rvsdg::Region & region, util::Graph * type
     auto & argument = *region.argument(n);
     AttachNodeOutput(node, argument, typeGraph);
 
+    // Give the argument a label using its local index, not the global argument index
+    // Use an uppercase A to distinguish
+    node.SetLabel(util::strfmt("a", n));
+
```
Comment (phate) [inline jlm/llvm/backend/dot/DotWriter.cpp:150]: It says "uppercase A" in the comment, but then a lowercase a is used.

Comment (phate) [reply jlm/llvm/backend/dot/DotWriter.cpp:150]: Also to distinguish it from what?

**Review by phate — DISMISSED** [review]

File: jlm/llvm/backend/dot/DotWriter.cpp (hunk 1 above)
Comment (haved) [reply jlm/llvm/backend/dot/DotWriter.cpp:150]: Good catch, this comment is not correct

**Comment by haved** [comment]
@phate I fixed the old comment, need another review

**Review by phate — APPROVED** [review]

**Pull request merged**